addons/mrp/models/mrp_production.py

```diff
@@ -2382,6 +2382,16 @@ def action_plan_with_components_availability(self):
     def _has_workorders(self):
         return self.workorder_ids
 
+    def unlink_old_bom_moves(self):
+        # Compares the BoM's lines to the MO's pre production pick.
+        pre_production_pick = self.picking_ids.filtered(lambda p: p.picking_type_id == self.warehouse_id.pbm_type_id)
+
+        if pre_production_pick:
+            pre_production_pick.ensure_one()
+            moves_to_unlink = pre_production_pick.move_ids.filtered(lambda m: not m.bom_line_id)
+            moves_to_unlink._action_cancel()
+            moves_to_unlink.unlink()
```

Comment:
You can have more than 1 `pre_production_pick` (e.g. backorders) [i.e. you'll get an error in this case]

This shouldn't apply to `done`/`picked` moves/pickings since you'd be erasing history

This method in general is overkill. You're going to risk removing manually added moves to the picking.